lib/rgeo/geographic/factory.rb

```diff
@@ -36,17 +36,25 @@ def initialize(impl_prefix, opts = {}) # :nodoc:
         @coordinate_dimension += 1 if @support_m
         @spatial_dimension = @support_z ? 3 : 2
 
-        @srid = (opts[:srid] || 4326).to_i
-        @proj4 = opts[:proj4]
-        if @proj4 && CoordSys.check!(:proj4)
-          if @proj4.is_a?(String) || @proj4.is_a?(Hash)
-            @proj4 = CoordSys::Proj4.create(@proj4)
-          end
-        end
         @coord_sys = opts[:coord_sys]
+        coord_sys_class = opts[:coord_sys_class]
+        unless coord_sys_class.is_a?(Class)
```

Comment:
```suggestion
        unless coord_sys_class.respond_to?(:create_from_wkt)
```

more flexible if someone wants to wrap the class ?